test/test/jfr/JfrTests.java

```diff
@@ -70,8 +70,8 @@ public void parseMultiModeRecording(TestProcess p) throws Exception {
             }
         }
 
-        Assert.isGreater(eventsCount.get("jdk.ExecutionSample"), 50);
-        Assert.isGreater(eventsCount.get("jdk.JavaMonitorEnter"), 50);
-        Assert.isGreater(eventsCount.get("jdk.ObjectAllocationInNewTLAB"), 50);
+        Assert.isGreater(eventsCount.get("jdk.ExecutionSample"), 50, "jdk.ExecutionSample > 50");
```

Comment:
I'm not a fan of repeating assertion in a message every time - this is more of a boilerplate that does not add much value. It's ok to leave it without a message. Or, given that stacktrace already contains line numbers, we may extract a line from the source code and show it to a user. Not sure it's worth doing, though.

Reply:
Added source parser, assertion failures looks like this now:

```
java.lang.AssertionError: Expected 0.13666666666666666 < 0.01
at test/test/recovery/RecoveryTests.java:42
:41    Output out = p.profile("-d 3 -e cpu -o collapsed");
:42 👉 Assert.isLess(out.ratio("vtable stub"), 0.01);
:43    Assert.isGreater(out.ratio("Numbers.loop"), 0.8);
at one.profiler.test.Assert.assertComparison(Assert.java:44)
at one.profiler.test.Assert.isLess(Assert.java:81)
at test.recovery.RecoveryTests.numbers(RecoveryTests.java:42)
```

before:

```
java.lang.AssertionError: Expected 0.13666666666666666 < 0.01
at one.profiler.test.Assert.assertComparison(Assert.java:44)
at one.profiler.test.Assert.isLess(Assert.java:81)
at test.recovery.RecoveryTests.numbers(RecoveryTests.java:42)
```